enigma/src/main/java/org/quiltmc/enigma/impl/plugin/RecordGetterFindingVisitor.java

```diff
@@ -0,0 +1,116 @@
+	}
+
+	@Override
+	public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+		super.visit(version, access, name, signature, superName, interfaces);
+		this.clazz = (access & Opcodes.ACC_RECORD) != 0 ? new ClassEntry(name) : null;
+		this.recordComponents.clear();
+	}
+
+	@Override
+	public RecordComponentVisitor visitRecordComponent(final String name, final String descriptor, final String signature) {
+		this.recordComponents.add(new RecordComponentNode(this.api, name, descriptor, signature));
+		return super.visitRecordComponent(name, descriptor, signature);
+	}
+
+	@Override
+	public FieldVisitor visitField(final int access, final String name, final String descriptor, final String signature, final Object value) {
+		if (this.clazz != null && ((access & Opcodes.ACC_PRIVATE) != 0) && this.recordComponents.stream().anyMatch(component -> component.name.equals(name))) {
+			FieldNode node = new FieldNode(this.api, access, name, descriptor, signature, value);
+			this.fields.add(node);
+			return node;
+		}
+
+		return super.visitField(access, name, descriptor, signature, value);
+	}
+
+	@Override
+	public MethodVisitor visitMethod(final int access, final String name, final String descriptor, final String signature, final String[] exceptions) {
+		if (this.clazz != null && ((access & Opcodes.ACC_PUBLIC) != 0)) {
+			MethodNode node = new MethodNode(this.api, access, name, descriptor, signature, exceptions);
+			this.methods.add(node);
+			return node;
+		}
+
+		return super.visitMethod(access, name, descriptor, signature, exceptions);
+	}
+
+	@Override
+	public void visitEnd() {
+		super.visitEnd();
+		try {
+			if (this.clazz != null) {
+				this.collectResults();
+			}
+		} catch (Exception ex) {
+			throw new RuntimeException(ex);
+		}
+	}
+
+	private void collectResults() {
+		for (RecordComponentNode component : this.recordComponents) {
+			FieldNode field = null;
+			for (FieldNode node : this.fields) {
+				if (node.name.equals(component.name) && node.desc.equals(component.descriptor)) {
+					field = node;
+					break;
+				}
+			}
+
+			if (field == null) {
+				throw new RuntimeException("Field not found for record component: " + component.name);
+			}
+
+			for (MethodNode method : this.methods) {
+				InsnList instructions = method.instructions;
+
+				// match bytecode to exact expected bytecode for a getter
+				// only check important instructions (ignore new frame instructions, etc.)
+				if (instructions.size() == 6
+						&& instructions.get(2).getOpcode() == Opcodes.ALOAD
+						&& instructions.get(3) instanceof FieldInsnNode fieldInsn
+						&& fieldInsn.getOpcode() == Opcodes.GETFIELD
+						&& fieldInsn.owner.equals(this.clazz.getName())
+						&& fieldInsn.desc.equals(field.desc)
+						&& fieldInsn.name.equals(field.name)
+						&& instructions.get(4).getOpcode() >= 172 && instructions.get(4).getOpcode() <= 177) { // return opcodes are 172 - 177
```

Comment:
```suggestion
						&& instructions.get(4).getOpcode() >= Opcodes.IRETURN && instructions.get(4).getOpcode() <= Opcodes.ARETURN) {
```
`RETURN` (177) is a void return, so it should check up to `ARETURN` (176)